File: frontend/src/types/crm.ts
export interface Tag {
  id: string;
  name: string;
  color?: string;
  created_at: string;
  updated_at: string;
}

export interface Account {
  id: string;
  name: string;
  industry?: string;
  website?: string;
  phone?: string;
  address_line1?: string;
  address_line2?: string;
  city?: string;
  state?: string;
  zip_code?: string;
  country?: string;
  annual_revenue?: number;
  employees_count?: number;
  description?: string;
  created_at: string;
  updated_at: string;
  created_by?: string;
  tags: Tag[];
}

export interface Contact {
  id: string;
  account_id?: string;
  first_name: string;
  last_name: string;
  email?: string;
  phone?: string;
  job_title?: string;
  status: string; // active, inactive, bounce, unsubscribed
  description?: string;
  created_at: string;
  updated_at: string;
  created_by?: string;
  account_name?: string;
  tags: Tag[];
}

export interface Lead {
  id: string;
  first_name: string;
  last_name: string;
  email?: string;
  phone?: string;
  company_name?: string;
  status: string; // new, contacted, qualified, lost, converted
  source?: string;
  score: number;
  assigned_to?: string;
  description?: string;
  converted_at?: string;
  converted_contact_id?: string;
  converted_account_id?: string;
  created_at: string;
  updated_at: string;
  created_by?: string;
  tags: Tag[];
}

// Creation and Update types
export interface AccountCreateInput {
  name: string;
  industry?: string;
  website?: string;
  phone?: string;
  address_line1?: string;
  address_line2?: string;
  city?: string;
  state?: string;
  zip_code?: string;
  country?: string;
  annual_revenue?: number;
  employees_count?: number;
  description?: string;
  tag_names?: string[];
}

export interface AccountUpdateInput extends Partial<AccountCreateInput> {}

export interface ContactCreateInput {
  first_name: string;
  last_name: string;
  email?: string;
  phone?: string;
  job_title?: string;
  status?: string;
  description?: string;
  account_id?: string;
  tag_names?: string[];
}

export interface ContactUpdateInput extends Partial<ContactCreateInput> {}

export interface LeadCreateInput {
  first_name: string;
  last_name: string;
  email?: string;
  phone?: string;
  company_name?: string;
  status?: string;
  source?: string;
  assigned_to?: string;
  description?: string;
  tag_names?: string[];
}

export interface LeadUpdateInput extends Partial<LeadCreateInput> {}

export interface LeadConvertInput {
  create_account: boolean;
  create_contact: boolean;
  existing_account_id?: string;
}

export interface DuplicateWarning {
  type: 'email' | 'phone';
  message: string;
}

File: backend/app/schemas/crm.py
from datetime import datetime
from typing import List, Optional
import uuid
from pydantic import BaseModel, ConfigDict, EmailStr, Field

# ============================================
# TAG SCHEMAS
# ============================================
class TagBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=50)
    color: Optional[str] = Field(None, max_length=20)

class TagCreate(TagBase):
    pass

class TagRead(TagBase):
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

# ============================================
# ACCOUNT SCHEMAS
# ============================================
class AccountBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=255)
    industry: Optional[str] = Field(None, max_length=100)
    website: Optional[str] = Field(None, max_length=255)
    phone: Optional[str] = Field(None, max_length=50)
    address_line1: Optional[str] = Field(None, max_length=255)
    address_line2: Optional[str] = Field(None, max_length=255)
    city: Optional[str] = Field(None, max_length=100)
    state: Optional[str] = Field(None, max_length=100)
    zip_code: Optional[str] = Field(None, max_length=20)
    country: Optional[str] = Field(None, max_length=100)
    annual_revenue: Optional[float] = Field(None, ge=0)
    employees_count: Optional[int] = Field(None, ge=0)
    description: Optional[str] = None

class AccountCreate(AccountBase):
    tag_names: Optional[List[str]] = Field(default_factory=list)

class AccountUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=255)
    industry: Optional[str] = Field(None, max_length=100)
    website: Optional[str] = Field(None, max_length=255)
    phone: Optional[str] = Field(None, max_length=50)
    address_line1: Optional[str] = Field(None, max_length=255)
    address_line2: Optional[str] = Field(None, max_length=255)
    city: Optional[str] = Field(None, max_length=100)
    state: Optional[str] = Field(None, max_length=100)
    zip_code: Optional[str] = Field(None, max_length=20)
    country: Optional[str] = Field(None, max_length=100)
    annual_revenue: Optional[float] = Field(None, ge=0)
    employees_count: Optional[int] = Field(None, ge=0)
    description: Optional[str] = None
    tag_names: Optional[List[str]] = None

class AccountRead(AccountBase):
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime
    created_by: Optional[uuid.UUID] = None
    tags: List[TagRead] = Field(default_factory=list)

    model_config = ConfigDict(from_attributes=True)

# ============================================
# CONTACT SCHEMAS
# ============================================
class ContactBase(BaseModel):
    first_name: str = Field(..., min_length=1, max_length=150)
    last_name: str = Field(..., min_length=1, max_length=150)
    email: Optional[EmailStr] = None
    phone: Optional[str] = Field(None, max_length=50)
    job_title: Optional[str] = Field(None, max_length=100)
    status: str = Field("active", max_length=50)
    description: Optional[str] = None
    account_id: Optional[uuid.UUID] = None

class ContactCreate(ContactBase):
    tag_names: Optional[List[str]] = Field(default_factory=list)

class ContactUpdate(BaseModel):
    first_name: Optional[str] = Field(None, min_length=1, max_length=150)
    last_name: Optional[str] = Field(None, min_length=1, max_length=150)
    email: Optional[EmailStr] = None
    phone: Optional[str] = Field(None, max_length=50)
    job_title: Optional[str] = Field(None, max_length=100)
    status: Optional[str] = Field(None, max_length=50)
    description: Optional[str] = None
    account_id: Optional[uuid.UUID] = None
    tag_names: Optional[List[str]] = None

class ContactRead(ContactBase):
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime
    created_by: Optional[uuid.UUID] = None
    account_name: Optional[str] = None
    tags: List[TagRead] = Field(default_factory=list)

    model_config = ConfigDict(from_attributes=True)

# ============================================
# LEAD SCHEMAS
# ============================================
class LeadBase(BaseModel):
    first_name: str = Field(..., min_length=1, max_length=150)
    last_name: str = Field(..., min_length=1, max_length=150)
    email: Optional[EmailStr] = None
    phone: Optional[str] = Field(None, max_length=50)
    company_name: Optional[str] = Field(None, max_length=255)
    status: str = Field("new", max_length=50)
    source: Optional[str] = Field(None, max_length=100)
    assigned_to: Optional[uuid.UUID] = None
    description: Optional[str] = None

class LeadCreate(LeadBase):
    tag_names: Optional[List[str]] = Field(default_factory=list)

class LeadUpdate(BaseModel):
    first_name: Optional[str] = Field(None, min_length=1, max_length=150)
    last_name: Optional[str] = Field(None, min_length=1, max_length=150)
    email: Optional[EmailStr] = None
    phone: Optional[str] = Field(None, max_length=50)
    company_name: Optional[str] = Field(None, max_length=255)
    status: Optional[str] = Field(None, max_length=50)
    source: Optional[str] = Field(None, max_length=100)
    assigned_to: Optional[uuid.UUID] = None
    description: Optional[str] = None
    tag_names: Optional[List[str]] = None

class LeadRead(LeadBase):
    id: uuid.UUID
    score: int
    created_at: datetime
    updated_at: datetime
    created_by: Optional[uuid.UUID] = None
    converted_at: Optional[datetime] = None
    converted_contact_id: Optional[uuid.UUID] = None
    converted_account_id: Optional[uuid.UUID] = None
    tags: List[TagRead] = Field(default_factory=list)

    model_config = ConfigDict(from_attributes=True)

class LeadConvertRequest(BaseModel):
    create_account: bool = True
    create_contact: bool = True
    existing_account_id: Optional[uuid.UUID] = None
